mis_builder_total_committed_purchase/models/mis_total_committed_purchase.py

```diff
@@ -0,0 +1,56 @@
+# Copyright 2017 ACSONE SA/NV
+# Copyright 2022 Camptocamp SA (https://www.camptocamp.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models, tools
+from odoo.modules.module import get_module_resource
+
+
+class MisTotalCommittedPurchase(models.Model):
+
+    _name = "mis.total.committed.purchase"
+    _description = "MIS Total Purchase Commitment"
+    _auto = False
+
+    name = fields.Char()
+    analytic_account_id = fields.Many2one(
+        comodel_name="account.analytic.account", string="Analytic Account"
+    )
+    account_id = fields.Many2one(comodel_name="account.account", string="Account")
+    company_id = fields.Many2one(comodel_name="res.company", string="Company")
+    credit = fields.Float()
+    debit = fields.Float()
+    date = fields.Date()
+
+    analytic_tag_ids = fields.Many2many(
+        comodel_name="account.analytic.tag",
+        relation="mis_total_committed_purchase_tag_rel",
+        column1="mis_total_committed_purchase_id",
+        column2="account_analytic_tag_id",
+        string="Analytic Tags",
+    )
+
+    def init(self):
+
+        with open(
+            get_module_resource(
+                "mis_builder_total_committed_purchase",
+                "data",
+                "mis_total_committed_purchase.sql",
+            )
+        ) as f:
+            tools.drop_view_if_exists(self.env.cr, "mis_total_committed_purchase")
+            self.env.cr.execute(f.read())
+
+            with open(
+                get_module_resource(
+                    "mis_builder_total_committed_purchase",
+                    "data",
+                    "mis_total_committed_purchase_tag_rel.sql",
+                )
+            ) as f2:
+                # Create many2many relation for account.analytic.tag
+                tools.drop_view_if_exists(
+                    self.env.cr, "mis_total_committed_purchase_tag_rel"
+                )
+                self.env.cr.execute(f2.read())
```

Comment:
@Camille0907 any reason to have second initialization inside the first context manager? Why such indentation?

Reply:
I must admit that I don't know... I just stupidly copied what was done [here](https://github.com/OCA/mis-builder/blob/15.0/mis_builder_demo/models/mis_committed_purchase.py#L48) 
I didn't have time to investigate more...

Reply:
you didn't understand the question :smile: 
you've copied it like:
```
with open() as f:
    f.read()
    ...
    with open() as f:
        f2.read()
```

then I would simply expect:
```
with open() as f:
    f.read()
    ...
with open() as f:
    f2.read()
```

How I actually modified it.

Reply:
In the module I copied, the second `self.env.cr.execute` was inside the `with` statement of the `open()`
```
        with open(script) as f:
            f.read()
            ...
            self.env.cr.execute(<second sql query>)
```

I've just put the second SQL query in a file, as the first one, but kept the same indent